### PR TITLE
472 Manage support tool groups automatically with ground zero

### PR DESCRIPTION
**Pull request by liamtoozer**
# Has the DDL schema changed?
*Pull requests which include schema changes should be labelled with the `schema change` label for visibility*

*The [common entity model pom.xml](ssdc-rm-common-entity-model/pom.xml) file `project.version` must be bumped
appropriately if there are any common entity code changes*

* [ ] This PR includes DDL schema changes and has been labelled correctly, bumping to new schema
  version: 
* [ ] The POM has been updated with an appropriate version bump if required

# Motivation and Context

We can now make the support tool UI groups default, seeded data. This will allow us to manipulate them via code in future if needed.

# What has changed

* Support groups are now created with ground zero - combined support permissions into one file
* Added new patch & rollback for support groups' inserts - inserts are idempotent
* Support groups are now seeded data for dev images
* Updated `ground_zero_and_restore_user_pemissions.sh` to exclude seeded data as this was causing failures when running
* Changed support super user file name to reflect new behaviour

# How to test?

Check when bringing up the image with docker dev that it all works together nicely

# Links

https://trello.com/c/6SlV8MMl/472-manage-support-tool-groups-automatically-in-ddl-5

### CODE REVIEW

File: patches/200_add_support_permissions.sql [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+INSERT INTO casev3.user_group_permission (id, authorised_activity, group_id, survey_id) VALUES ('c484fdc8-3bfd-40ad-b608-75b9d94b434e', 'CREATE_CASE_INVALID_CASE', 'a25c7f99-d2ce-4267-aea4-0a133028f793', NULL) ON CONFLICT DO NOTHING;
+INSERT INTO casev3.user_group_permission (id, authorised_activity, group_id, survey_id) VALUES ('336002f6-b98b-40dc-acf6-6490b6348d8e', 'UPDATE_SAMPLE', 'a25c7f99-d2ce-4267-aea4-0a133028f793', NULL) ON CONFLICT DO NOTHING;
+INSERT INTO casev3.user_group_permission (id, authorised_activity, group_id, survey_id) VALUES ('20348a51-352f-4f2d-98d3-d6549110b9bf', 'UPDATE_SAMPLE_SENSITIVE', 'a25c7f99-d2ce-4267-aea4-0a133028f793', NULL) ON CONFLICT DO NOTHING;
+INSERT INTO casev3.user_group_permission (id, authorised_activity, group_id, survey_id) VALUES ('07d11c99-a41b-4039-b1e2-90d13b190783', 'EXCEPTION_MANAGER_PEEK', 'a25c7f99-d2ce-4267-aea4-0a133028f793', NULL) ON CONFLICT DO NOTHING;
+INSERT INTO casev3.user_group_permission (id, authorised_activity, group_id, survey_id) VALUES ('96dfd6f3-3692-4c03-8989-f19156854d36', 'EXCEPTION_MANAGER_QUARANTINE', 'a25c7f99-d2ce-4267-aea4-0a133028f793', NULL) ON CONFLICT DO NOTHING;
```
[the remark this answers is not in the document]
Comment (liamtoozer) [reply patches/200_add_support_permissions.sql:40]: Ah good spot 👍  Copied and pasted these from the originals and didn't notice the doubles. Will remove

File: patches/rollback/200_add_support_permissions.sql [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+DELETE FROM casev3.user_group_permission WHERE id = '5b34b96b-b813-4cab-af07-ea6e9c3c2d0d';
+DELETE FROM casev3.user_group_permission WHERE id = '6a4dd655-423d-4d56-bc38-604a61e55721';
+DELETE FROM casev3.user_group_permission WHERE id = '4b244c4d-d3ca-4841-a396-d03dc200396f';
+DELETE FROM casev3.user_group_permission WHERE id = '5798a4a2-9ab8-4509-9f88-71fde81e6948';
+DELETE FROM casev3.user_group_permission WHERE id = '5cfe474a-71f3-46bf-b9fe-fa6388c8c9da';
+DELETE FROM casev3.user_group_permission WHERE id = 'c484fdc8-3bfd-40ad-b608-75b9d94b434e';
+DELETE FROM casev3.user_group_permission WHERE id = '336002f6-b98b-40dc-acf6-6490b6348d8e';
+DELETE FROM casev3.user_group_permission WHERE id = '20348a51-352f-4f2d-98d3-d6549110b9bf';
+DELETE FROM casev3.user_group_permission WHERE id = '07d11c99-a41b-4039-b1e2-90d13b190783';
+DELETE FROM casev3.user_group_permission WHERE id = '96dfd6f3-3692-4c03-8989-f19156854d36';
```
[the remark this answers is not in the document]
Comment (liamtoozer) [reply patches/rollback/200_add_support_permissions.sql:37]: Ah good spot 👍  Copied and pasted these from the originals and didn't notice the doubles. Will remove

File: ui-permissions/RM-support-permissions.sql [hunk 3]
```diff
@@ -1,27 +1,67 @@
...
+INSERT INTO casev3.user_group_permission (id, authorised_activity, group_id, survey_id) VALUES ('c484fdc8-3bfd-40ad-b608-75b9d94b434e', 'CREATE_CASE_INVALID_CASE', 'a25c7f99-d2ce-4267-aea4-0a133028f793', NULL) ON CONFLICT DO NOTHING;
+INSERT INTO casev3.user_group_permission (id, authorised_activity, group_id, survey_id) VALUES ('336002f6-b98b-40dc-acf6-6490b6348d8e', 'UPDATE_SAMPLE', 'a25c7f99-d2ce-4267-aea4-0a133028f793', NULL) ON CONFLICT DO NOTHING;
+INSERT INTO casev3.user_group_permission (id, authorised_activity, group_id, survey_id) VALUES ('20348a51-352f-4f2d-98d3-d6549110b9bf', 'UPDATE_SAMPLE_SENSITIVE', 'a25c7f99-d2ce-4267-aea4-0a133028f793', NULL) ON CONFLICT DO NOTHING;
+INSERT INTO casev3.user_group_permission (id, authorised_activity, group_id, survey_id) VALUES ('07d11c99-a41b-4039-b1e2-90d13b190783', 'EXCEPTION_MANAGER_PEEK', 'a25c7f99-d2ce-4267-aea4-0a133028f793', NULL) ON CONFLICT DO NOTHING;
+INSERT INTO casev3.user_group_permission (id, authorised_activity, group_id, survey_id) VALUES ('96dfd6f3-3692-4c03-8989-f19156854d36', 'EXCEPTION_MANAGER_QUARANTINE', 'a25c7f99-d2ce-4267-aea4-0a133028f793', NULL) ON CONFLICT DO NOTHING;
```
[the remark this answers is not in the document]
Comment (liamtoozer) [reply ui-permissions/RM-support-permissions.sql:33]: Ah good spot 👍  Copied and pasted these from the originals and didn't notice the doubles. Will remove

File: dev-common-postgres-image/init.sql [hunk 4]
```diff
@@ -505,5 +505,74 @@ CREATE TABLE ddl_version.version (version_tag varchar(256) PRIMARY KEY, updated_
...
+INSERT INTO casev3.user_group_permission (id, authorised_activity, group_id, survey_id) VALUES ('c484fdc8-3bfd-40ad-b608-75b9d94b434e', 'CREATE_CASE_INVALID_CASE', 'a25c7f99-d2ce-4267-aea4-0a133028f793', NULL) ON CONFLICT DO NOTHING;
+INSERT INTO casev3.user_group_permission (id, authorised_activity, group_id, survey_id) VALUES ('336002f6-b98b-40dc-acf6-6490b6348d8e', 'UPDATE_SAMPLE', 'a25c7f99-d2ce-4267-aea4-0a133028f793', NULL) ON CONFLICT DO NOTHING;
+INSERT INTO casev3.user_group_permission (id, authorised_activity, group_id, survey_id) VALUES ('20348a51-352f-4f2d-98d3-d6549110b9bf', 'UPDATE_SAMPLE_SENSITIVE', 'a25c7f99-d2ce-4267-aea4-0a133028f793', NULL) ON CONFLICT DO NOTHING;
+INSERT INTO casev3.user_group_permission (id, authorised_activity, group_id, survey_id) VALUES ('07d11c99-a41b-4039-b1e2-90d13b190783', 'EXCEPTION_MANAGER_PEEK', 'a25c7f99-d2ce-4267-aea4-0a133028f793', NULL) ON CONFLICT DO NOTHING;
+INSERT INTO casev3.user_group_permission (id, authorised_activity, group_id, survey_id) VALUES ('96dfd6f3-3692-4c03-8989-f19156854d36', 'EXCEPTION_MANAGER_QUARANTINE', 'a25c7f99-d2ce-4267-aea4-0a133028f793', NULL) ON CONFLICT DO NOTHING;
```
[the remark this answers is not in the document]
Comment (liamtoozer) [reply dev-common-postgres-image/init.sql:544]: Ah good spot 👍  Copied and pasted these from the originals and didn't notice the doubles. Will remove

**Review by gavinleeedwards — APPROVED** [review]
Looks good 👍

**Review by LukeLoze1 — APPROVED** [review]
works for me

**Pull request merged**